### ISSUE TITLE
memory leak in rank 0 broker while jobs are running

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#607]
Here is another run of a bunch of 8 node jobs back to back similar to #604, this time tracking the rank 0 broker RSS.  There must be a memory leak here unless this rss growth can be explained by kvs/content caching or fragmentation.
```
#!/bin/bash
size=$(flux getattr size)
echo $FLUX_URI
for i in `seq 1 4800`; do
    elapsed=$(/usr/bin/time -f "%e" flux wreckrun --ntasks $size /bin/true 2>&1)
    broker_rss=$(flux comms-stats --rusage cmb --parse maxrss)
    flux logger --priority soak.info $i $elapsed $broker_rss
done
```
![soak](https://cloud.githubusercontent.com/assets/169947/13793616/003c46dc-eab5-11e5-80b7-8c0b3f0f2e9d.png)

**Comment by garlick** [comment]
With two fixes, getting better as shown in this partial run:
* kvs internal commit matchtag leak
* keepalive message leak

![soak](https://cloud.githubusercontent.com/assets/169947/13862717/35216d34-ec52-11e5-9831-659b3d77dcd6.png)

**Comment by garlick** [comment]
This is solved for the most part by #610, #614, #617.
